infoblox_client/utils.py

```diff
@@ -35,16 +34,14 @@ def is_valid_ip(ip):
 def generate_duid(mac):
     """DUID is consisted of 10 hex numbers.
 
-    0x00 + 3 random hex + mac with 6 hex
+    0x00 + mac with last 3 hex + mac with 6 hex
     """
     valid = mac and isinstance(mac, six.string_types)
     if not valid:
-        ValueError("Invalid argument was passed")
-    duid = [0x00,
-            random.randint(0x00, 0x7f),
-            random.randint(0x00, 0xff),
-            random.randint(0x00, 0xff)]
-    return ':'.join(map(lambda x: "%02x" % x, duid)) + ':' + mac
+        raise ValueError("Invalid argument was passed")
+    duid = ['00']
+    duid.extend(mac.split(':')[-3:])
+    return ':'.join(map(lambda x: "%s" % x, duid)) + ':' + mac
```

Comment:
@adsri , thanks for your explanation!

I don't have a strong opinion about this, it just seems unnecessarily complicated.

I also wanted to point out that, since you have already clearly described what this method is doing (which is good!), I think the simplified version is fairly obvious too.

I'll leave it up to you.